source/change-streams.rst

```diff
@@ -41,49 +41,12 @@ Terms
 Resumable Error
 ^^^^^^^^^^^^^^^
 
-An error is considered resumable if it meets any of the following criteria:
+An error on a getMore command is considered resumable if it meets any of the following criteria:
 
 - any error encountered which is not a server error (e.g. a timeout error or
   network error)
 
-- a server error response with any of the following error codes:
-
-  .. list-table::
-    :header-rows: 1
-
-    * - Error Name
-      - Error Code
-    * - HostUnreachable
-      - 6
-    * - HostNotFound
-      - 7
-    * - NetworkTimeout
-      - 89
-    * - SocketException
-      - 9001
-    * - NotMaster
-      - 10107
-    * - NotMasterNoSlaveOk
-      - 13435
-    * - NotMasterOrSecondary
-      - 13436
-    * - InterruptedDueToReplStateChange
-      - 11602
-    * - PrimarySteppedDown
-      - 189
-    * - ExceededTimeLimit
-      - 50
-    * - RetryChangeStream
-      - 234
-    * - ElectionInProgress
-      - 216
-
-- a server error response without an error code or one different from those
-  listed above, but with an error message containing the substring "not
-  master" or "node is recovering"
-
-- *any* server error response from a getMore command excluding those
-  containing the following error codes
+- any server error response containing an error code *excluding* the following
```

Comment:
What about:

- CappedPositionLost
- Interrupted

The C# driver resumes on those also though I no longer remember why.

Reply:
CappedPositionLost should not be resumed because the resume attempt will also fail. This error occurs when the change stream cursor falls off the end of the oplog. The resume attempt would fail complaining that the resumeToken is too old.

Interrupted should not be resumed because that is the error code received when running killOp. If we resumed after Interrupted then change stream cursors would be unkillable.

Reply:
I think the argument for excluding them is that CappedPositionLost means a resume is hopeless since we've fallen off of the oplog, and that Interrupted occurs when the user intentionally kills the operation, so presumably they do not want them to attempt resuming.